pkgs/games/starsector/default.nix

```diff
@@ -42,36 +44,47 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  # need to cd into $out in order for classpath to pick up correct jar files
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin
+    mkdir -p $out/bin $out/share/starsector
     rm -r jre_linux # remove jre7
     rm starfarer.api.zip
-    cp -r ./* $out
+    cp -r ./* $out/share/starsector
 
     mkdir -p $out/share/icons/hicolor/64x64/apps
-    ln -s $out/graphics/ui/s_icon64.png $out/share/icons/hicolor/64x64/apps/starsector.png
+    ln -s $out/share/starsector/graphics/ui/s_icon64.png $out/share/icons/hicolor/64x64/apps/starsector.png
 
-    wrapProgram $out/starsector.sh \
+    # Replace the upstream launch script with one that doesn't try to write to $out at runtime
+    cat << EOF > $out/share/starsector/starsector.sh
+    #!${stdenv.shell}
+    XDG_DATA_HOME=\''${XDG_DATA_HOME:="\$HOME/.local/share"}
+    mkdir -p \$XDG_DATA_HOME/starsector
+    cd $out/share/starsector
+    java -server \\
+      -XX:CompilerThreadPriority=1 \\
+      -XX:+CompilerThreadHintNoPreempt \\
+      -Djava.library.path=$out/share/starsector/native/linux \\
+      -Xms1536m \\
+      -Xmx1536m \\
```

Comment:
Maybe make memory settings overridable?
For people with heaps of mods to not have to patch this patch